### ISSUE TITLE
Add logging of resulting webpack loader objects during invalid configuration object errors

### ISSUE THREAD

**Issue by kylemh** [vercel/next.js#17827]
# Feature request

## Is your feature request related to a problem? Please describe.

Webpack 5 is out and I was excited to try using it! I've pinned the dependency via the `resolutions` field in `package.json`, but - upon running `next build` - I'm met with a large callstack related to: "ValidationError: Invalid configuration object.". The problem is that the Webpack config is defined as a result of many functions in next source code AND the amalgamation of many plugins (or not if one doesn't use them.)

It would be helpful for both myself and plugin owners, if the specific loader in question (and it's shape/appearance) was brought to the attention of developers when facing a webpack build validation error.

## Describe the solution you'd like

If a webpack build validation error throws, the problematic loaders should be logged.

**Comment by ScriptedAlchemy** [comment]
whats the config object saying. Show me the error it emits. 
also yarn list webpack and make sure there's not more than once being installed

**Comment by kylemh** [comment]
```sh
$ next build
info  - Loaded env from /Users/kyle/Code/@air/next/.env.local
warn  - You have enabled experimental feature(s).
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use them at your own risk.


> Build error occurred
ValidationError: Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration[0].module.rules[4] should be one of these:
   ["..." | object { compiler?, dependency?, descriptionData?, enforce?, exclude?, generator?, include?, issuer?, loader?, mimetype?, oneOf?, options?, parser?, realResource?, resolve?, resource?, resourceFragment?, resourceQuery?, rules?, sideEffects?, test?, type?, use? }, ...]
   -> A rule.
   Details:
    * configuration[0].module.rules[4].loader should be a non-empty string.
      -> A loader request.
 - configuration[0].output has an unknown property 'ecmaVersion'. These properties are valid:
   object { assetModuleFilename?, auxiliaryComment?, charset?, chunkFilename?, chunkFormat?, chunkLoadTimeout?, chunkLoading?, chunkLoadingGlobal?, compareBeforeEmit?, crossOriginLoading?, devtoolFallbackModuleFilenameTemplate?, devtoolModuleFilenameTemplate?, devtoolNamespace?, enabledChunkLoadingTypes?, enabledLibraryTypes?, enabledWasmLoadingTypes?, environment?, filename?, globalObject?, hashDigest?, hashDigestLength?, hashFunction?, hashSalt?, hotUpdateChunkFilename?, hotUpdateGlobal?, hotUpdateMainFilename?, iife?, importFunctionName?, importMetaName?, library?, libraryExport?, libraryTarget?, module?, path?, pathinfo?, publicPath?, scriptType?, sourceMapFilename?, sourcePrefix?, strictModuleExceptionHandling?, umdNamedDefine?, uniqueName?, wasmLoading?, webassemblyModuleFilename?, workerChunkLoading?, workerWasmLoading? }
   -> Options affecting the output of the compilation. `output` options tell webpack how to write the compiled files to disk.
   Did you mean output.environment (output.ecmaVersion was a temporary configuration option during webpack 5 beta)?
 - configuration[0].watchOptions.ignored[2] should be a non-empty string.
   -> A glob pattern for files that should be ignored from watching.
 - configuration[1].module.rules[4] should be one of these:
   ["..." | object { compiler?, dependency?, descriptionData?, enforce?, exclude?, generator?, include?, issuer?, loader?, mimetype?, oneOf?, options?, parser?, realResource?, resolve?, resource?, resourceFragment?, resourceQuery?, rules?, sideEffects?, test?, type?, use? }, ...]
   -> A rule.
   Details:
    * configuration[1].module.rules[4].loader should be a non-empty string.
      -> A loader request.
 - configuration[1].output has an unknown property 'ecmaVersion'. These properties are valid:
   object { assetModuleFilename?, auxiliaryComment?, charset?, chunkFilename?, chunkFormat?, chunkLoadTimeout?, chunkLoading?, chunkLoadingGlobal?, compareBeforeEmit?, crossOriginLoading?, devtoolFallbackModuleFilenameTemplate?, devtoolModuleFilenameTemplate?, devtoolNamespace?, enabledChunkLoadingTypes?, enabledLibraryTypes?, enabledWasmLoadingTypes?, environment?, filename?, globalObject?, hashDigest?, hashDigestLength?, hashFunction?, hashSalt?, hotUpdateChunkFilename?, hotUpdateGlobal?, hotUpdateMainFilename?, iife?, importFunctionName?, importMetaName?, library?, libraryExport?, libraryTarget?, module?, path?, pathinfo?, publicPath?, scriptType?, sourceMapFilename?, sourcePrefix?, strictModuleExceptionHandling?, umdNamedDefine?, uniqueName?, wasmLoading?, webassemblyModuleFilename?, workerChunkLoading?, workerWasmLoading? }
   -> Options affecting the output of the compilation. `output` options tell webpack how to write the compiled files to disk.
   Did you mean output.environment (output.ecmaVersion was a temporary configuration option during webpack 5 beta)?
 - configuration[1].watchOptions.ignored[2] should be a non-empty string.
   -> A glob pattern for files that should be ignored from watching.
    at validate (/Users/kyle/Code/@air/next/node_modules/webpack/node_modules/schema-utils/dist/validate.js:104:11)
    at validateSchema (/Users/kyle/Code/@air/next/node_modules/webpack/lib/validateSchema.js:73:2)
    at create (/Users/kyle/Code/@air/next/node_modules/webpack/lib/webpack.js:102:3)
    at webpack (/Users/kyle/Code/@air/next/node_modules/webpack/lib/webpack.js:139:31)
    at f (/Users/kyle/Code/@air/next/node_modules/webpack/lib/index.js:35:15)
    at /Users/kyle/Code/@air/next/node_modules/next/dist/build/compiler.js:4:172
    at new Promise (<anonymous>)
    at runCompiler (/Users/kyle/Code/@air/next/node_modules/next/dist/build/compiler.js:4:106)
    at build (/Users/kyle/Code/@air/next/node_modules/next/dist/build/index.js:13:348) {
  errors: [
    {
      keyword: 'oneOf',
      dataPath: '[0].module.rules',
      schemaPath: '#/properties/rules/oneOf',
      params: [Object],
      message: 'should match exactly one schema in oneOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    },
    {
      keyword: 'additionalProperties',
      dataPath: '[0].output',
      schemaPath: '#/additionalProperties',
      params: [Object],
      message: 'should NOT have additional properties',
      schema: false,
      parentSchema: [Object],
      data: [Object]
    },
    {
      keyword: 'anyOf',
      dataPath: '[0].watchOptions.ignored',
      schemaPath: '#/definitions/WatchOptions/properties/ignored/anyOf',
      params: {},
      message: 'should match some schema in anyOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    },
    {
      keyword: 'oneOf',
      dataPath: '[1].module.rules',
      schemaPath: '#/properties/rules/oneOf',
      params: [Object],
      message: 'should match exactly one schema in oneOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    },
    {
      keyword: 'additionalProperties',
      dataPath: '[1].output',
      schemaPath: '#/additionalProperties',
      params: [Object],
      message: 'should NOT have additional properties',
      schema: false,
      parentSchema: [Object],
      data: [Object]
    },
    {
      keyword: 'anyOf',
      dataPath: '[1].watchOptions.ignored',
      schemaPath: '#/definitions/WatchOptions/properties/ignored/anyOf',
      params: {},
      message: 'should match some schema in anyOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    }
  ],
  schema: {
    definitions: {
      Amd: [Object],
      AssetModuleFilename: [Object],
      AuxiliaryComment: [Object],
      Bail: [Object],
      CacheOptions: [Object],
      CacheOptionsNormalized: [Object],
      Charset: [Object],
      ChunkFilename: [Object],
      ChunkFormat: [Object],
      ChunkLoadTimeout: [Object],
      ChunkLoading: [Object],
      ChunkLoadingGlobal: [Object],
      ChunkLoadingType: [Object],
      CompareBeforeEmit: [Object],
      Context: [Object],
      CrossOriginLoading: [Object],
      Dependencies: [Object],
      DevServer: [Object],
      DevTool: [Object],
      DevtoolFallbackModuleFilenameTemplate: [Object],
      DevtoolModuleFilenameTemplate: [Object],
      DevtoolNamespace: [Object],
      EnabledChunkLoadingTypes: [Object],
      EnabledLibraryTypes: [Object],
      EnabledWasmLoadingTypes: [Object],
      Entry: [Object],
      EntryDescription: [Object],
      EntryDescriptionNormalized: [Object],
      EntryDynamic: [Object],
      EntryDynamicNormalized: [Object],
      EntryItem: [Object],
      EntryNormalized: [Object],
      EntryObject: [Object],
      EntryRuntime: [Object],
      EntryStatic: [Object],
      EntryStaticNormalized: [Object],
      EntryUnnamed: [Object],
      Environment: [Object],
      Experiments: [Object],
      ExternalItem: [Object],
      Externals: [Object],
      ExternalsPresets: [Object],
      ExternalsType: [Object],
      FileCacheOptions: [Object],
      Filename: [Object],
      FilterItemTypes: [Object],
      FilterTypes: [Object],
      GlobalObject: [Object],
      HashDigest: [Object],
      HashDigestLength: [Object],
      HashFunction: [Object],
      HashSalt: [Object],
      HotUpdateChunkFilename: [Object],
      HotUpdateGlobal: [Object],
      HotUpdateMainFilename: [Object],
      IgnoreWarnings: [Object],
      IgnoreWarningsNormalized: [Object],
      Iife: [Object],
      ImportFunctionName: [Object],
      ImportMetaName: [Object],
      InfrastructureLogging: [Object],
      Library: [Object],
      LibraryCustomUmdCommentObject: [Object],
      LibraryCustomUmdObject: [Object],
      LibraryExport: [Object],
      LibraryName: [Object],
      LibraryOptions: [Object],
      LibraryType: [Object],
      Loader: [Object],
      MemoryCacheOptions: [Object],
      Mode: [Object],
      ModuleOptions: [Object],
      Name: [Object],
      Node: [Object],
      NodeOptions: [Object],
      Optimization: [Object],
      OptimizationRuntimeChunk: [Object],
      OptimizationRuntimeChunkNormalized: [Object],
      OptimizationSplitChunksCacheGroup: [Object],
      OptimizationSplitChunksGetCacheGroups: [Object],
      OptimizationSplitChunksOptions: [Object],
      OptimizationSplitChunksSizes: [Object],
      Output: [Object],
      OutputModule: [Object],
      OutputNormalized: [Object],
      Parallelism: [Object],
      Path: [Object],
      Pathinfo: [Object],
      Performance: [Object],
      PerformanceOptions: [Object],
      Plugins: [Object],
      Profile: [Object],
      PublicPath: [Object],
      RecordsInputPath: [Object],
      RecordsOutputPath: [Object],
      RecordsPath: [Object],
      Resolve: [Object],
      ResolveAlias: [Object],
      ResolveLoader: [Object],
      ResolveOptions: [Object],
      ResolvePluginInstance: [Object],
      RuleSetCondition: [Object],
      RuleSetConditionAbsolute: [Object],
      RuleSetConditionOrConditions: [Object],
      RuleSetConditionOrConditionsAbsolute: [Object],
      RuleSetConditions: [Object],
      RuleSetConditionsAbsolute: [Object],
      RuleSetLoader: [Object],
      RuleSetLoaderOptions: [Object],
      RuleSetRule: [Object],
      RuleSetRules: [Object],
      RuleSetUse: [Object],
      RuleSetUseItem: [Object],
      ScriptType: [Object],
      SnapshotOptions: [Object],
      SourceMapFilename: [Object],
      SourcePrefix: [Object],
      StatsOptions: [Object],
      StatsValue: [Object],
      StrictModuleExceptionHandling: [Object],
      Target: [Object],
      UmdNamedDefine: [Object],
      UniqueName: [Object],
      WasmLoading: [Object],
      WasmLoadingType: [Object],
      Watch: [Object],
      WatchOptions: [Object],
      WebassemblyModuleFilename: [Object],
      WebpackOptionsNormalized: [Object],
      WebpackPluginFunction: [Object],
      WebpackPluginInstance: [Object]
    },
    description: 'Options object as provided by the user.',
    type: 'object',
    additionalProperties: false,
    properties: {
      amd: [Object],
      bail: [Object],
      cache: [Object],
      context: [Object],
      dependencies: [Object],
      devServer: [Object],
      devtool: [Object],
      entry: [Object],
      experiments: [Object],
      externals: [Object],
      externalsPresets: [Object],
      externalsType: [Object],
      ignoreWarnings: [Object],
      infrastructureLogging: [Object],
      loader: [Object],
      mode: [Object],
      module: [Object],
      name: [Object],
      node: [Object],
      optimization: [Object],
      output: [Object],
      parallelism: [Object],
      performance: [Object],
      plugins: [Object],
      profile: [Object],
      recordsInputPath: [Object],
      recordsOutputPath: [Object],
      recordsPath: [Object],
      resolve: [Object],
      resolveLoader: [Object],
      snapshot: [Object],
      stats: [Object],
      target: [Object],
      watch: [Object],
      watchOptions: [Object]
    }
  },
  headerName: 'Webpack',
  baseDataPath: 'configuration',
  postFormatter: [Function: postFormatter]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by kylemh** [comment]
Lemme know if I can provide any more context @ScriptedAlchemy - really excited to take advantage of your hard work.

**Comment by ScriptedAlchemy** [comment]
Looks like an old version of next. Make sure you're on the latest copy. The ecma feature set was changed around the RC release I think. 

Yarn list next and webpack. There must be an old copy of next. Also try clearing lock and module modules.

**Comment by kylemh** [comment]
![Screen Shot 2020-10-12 at 9 29 11 PM](https://user-images.githubusercontent.com/9523719/95815681-0f0b5b00-0cd2-11eb-970f-7dd02d8037a7.png)

No other versions of Next.js - is the suggestion to use a canary version? I tried clearing node_modules to no avail.

**Comment by kylemh** [comment]
I didn't think an upgrade from Next.js 9.5.3 -> 9.5.5 was necessary for Webpack, but I upgraded anyways and received the same error.

**Comment by ScriptedAlchemy** [comment]
Looks like webpack dev server configs are old, this changed a little. Do you use next transpile modules? Or have you configured any specific watch options. 

Webpack stable release might have changed some final things. I've had some issues when jumping from beta 31 to RC1 then to stable. 

9.5.5 might have made final refinements. What we are likely seeing right now is a plugin that's it yet been updated. I've had this exact issue. For me it was next-transpile-modules which sets watchOptions. That changed slightly. But I did PR that project & updated it to work, also published a fork. 

Have you checked loaders and plugins are up to date? Typically there's not too much that's changed. But there's a few little things, like this error which needs updating. If you want find where watchOptions is coming from, do a global search in your repo and exclude webpack. Then we would have a better idea of what sets watch options from a third party plugin.

**Comment by ScriptedAlchemy** [comment]
Your lock file shows another version of webpack. V4 In there. I also see virtual modules plugin which is not working (broke svelte-loader)

**Comment by kylemh** [comment]
The lockfile shows multiple versions of webpack resolving to one (v5). Not that all the versions are installed.

I do use next transpile modules! I listed my next.config.js higher up, assuming that one of the plugins I leveraged was at fault. That's why the original issue was about logging so I can find out the actual problematic loader.

**Comment by ScriptedAlchemy** [comment]
Yeah it's transpile modules. Use my copy till the author updates it. Mine is a little slower because I turn off caching on transpiled modules to make it work. 

It's not recommended ofc. But it works and would unblock you at the expense of slightly slower builds. Had the fork in production for about  a month or two. I pushed a bunch of new fixed to it today as I've started updating all my apps to v5 stable 

https://www.npmjs.com/%40module-federation%2Fnext-transpile-modules

I won't be maintaining the fork once the main repo is updated. So keep an eye transpile-modules

**Comment by kylemh** [comment]
Awesome thanks! I still think the original point would be helpful for debug purposes, so I'll leave this issue open

**Comment by ScriptedAlchemy** [comment]
Haha sure. Just unblocking you 👍

**Comment by nandorojo** [comment]
Hey @ScriptedAlchemy. First off, thanks so much for your amazing work with Webpack.

I'm having a similar in my next app (`next@10.1.4-canary.2`). I'm failing to get Webpack 5 to work, and I'm pulling my hair out a bit, since I'm not that familiar with how Webpack works.

I've been using your fork of `next-transpile-modules` in my monorepo, since I had trouble using the main one. I'm now trying to upgrade to webpack 5.

- `"next-transpile-modules": "^6.4.0"`

However, I'm getting this error:

<details>
<summary>Terminal error</summary>

```sh
yarn build
yarn run v1.22.5
$ next build
info  - Using webpack 5. Reason: future.webpack5 option enabled https://nextjs.org/docs/messages/webpack5
info  - Checking validity of types

> Build error occurred
ValidationError: Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration[0].module.rules[3] should be one of these:
   ["..." | object { compiler?, dependency?, descriptionData?, enforce?, exclude?, generator?, include?, issuer?, issuerLayer?, layer?, loader?, mimetype?, oneOf?, options?, parser?, realResource?, resolve?, resource?, resourceFragment?, resourceQuery?, rules?, sideEffects?, test?, type?, use? }, ...]
   -> A rule.
   Details:
    * configuration[0].module.rules[3].loader should be a non-empty string.
      -> A loader request.
 - configuration[0].watchOptions.ignored[3] should be a non-empty string.
   -> A glob pattern for files that should be ignored from watching.
 - configuration[1].module.rules[3] should be one of these:
   ["..." | object { compiler?, dependency?, descriptionData?, enforce?, exclude?, generator?, include?, issuer?, issuerLayer?, layer?, loader?, mimetype?, oneOf?, options?, parser?, realResource?, resolve?, resource?, resourceFragment?, resourceQuery?, rules?, sideEffects?, test?, type?, use? }, ...]
   -> A rule.
   Details:
    * configuration[1].module.rules[3].loader should be a non-empty string.
      -> A loader request.
 - configuration[1].watchOptions.ignored[3] should be a non-empty string.
   -> A glob pattern for files that should be ignored from watching.
    at validate (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/compiled/schema-utils3/index.js:1:153657)
    at validateSchema (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/compiled/webpack/bundle5.js:131116:2)
    at create (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/compiled/webpack/bundle5.js:134538:4)
    at webpack (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/compiled/webpack/bundle5.js:134575:32)
    at f (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/compiled/webpack/bundle5.js:92764:15)
    at /Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/build/compiler.js:4:172
    at new Promise (<anonymous>)
    at runCompiler (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/build/compiler.js:4:106)
    at /Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/build/index.js:15:418
    at Span.traceAsyncFn (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/telemetry/trace/trace.js:5:590)
    at /Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/build/index.js:15:376
    at async Span.traceAsyncFn (/Users/fernandorojo/Developer/madison-green/beatgig-playground/node_modules/next/dist/telemetry/trace/trace.js:5:584) {
  errors: [
    {
      keyword: 'oneOf',
      dataPath: '[0].module.rules',
      schemaPath: '#/properties/rules/oneOf',
      params: [Object],
      message: 'should match exactly one schema in oneOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    },
    {
      keyword: 'anyOf',
      dataPath: '[0].watchOptions.ignored',
      schemaPath: '#/definitions/WatchOptions/properties/ignored/anyOf',
      params: {},
      message: 'should match some schema in anyOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    },
    {
      keyword: 'oneOf',
      dataPath: '[1].module.rules',
      schemaPath: '#/properties/rules/oneOf',
      params: [Object],
      message: 'should match exactly one schema in oneOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    },
    {
      keyword: 'anyOf',
      dataPath: '[1].watchOptions.ignored',
      schemaPath: '#/definitions/WatchOptions/properties/ignored/anyOf',
      params: {},
      message: 'should match some schema in anyOf',
      schema: [Array],
      parentSchema: [Object],
      data: [Array],
      children: [Array]
    }
  ],
  schema: {
    definitions: {
      Amd: [Object],
      AssetFilterItemTypes: [Object],
      AssetFilterTypes: [Object],
      AssetGeneratorDataUrl: [Object],
      AssetGeneratorDataUrlFunction: [Object],
      AssetGeneratorDataUrlOptions: [Object],
      AssetGeneratorOptions: [Object],
      AssetInlineGeneratorOptions: [Object],
      AssetModuleFilename: [Object],
      AssetParserDataUrlFunction: [Object],
      AssetParserDataUrlOptions: [Object],
      AssetParserOptions: [Object],
      AssetResourceGeneratorOptions: [Object],
      AuxiliaryComment: [Object],
      Bail: [Object],
      CacheOptions: [Object],
      CacheOptionsNormalized: [Object],
      Charset: [Object],
      ChunkFilename: [Object],
      ChunkFormat: [Object],
      ChunkLoadTimeout: [Object],
      ChunkLoading: [Object],
      ChunkLoadingGlobal: [Object],
      ChunkLoadingType: [Object],
      Clean: [Object],
      CleanOptions: [Object],
      CompareBeforeEmit: [Object],
      Context: [Object],
      CrossOriginLoading: [Object],
      Dependencies: [Object],
      DevServer: [Object],
      DevTool: [Object],
      DevtoolFallbackModuleFilenameTemplate: [Object],
      DevtoolModuleFilenameTemplate: [Object],
      DevtoolNamespace: [Object],
      EmptyGeneratorOptions: [Object],
      EmptyParserOptions: [Object],
      EnabledChunkLoadingTypes: [Object],
      EnabledLibraryTypes: [Object],
      EnabledWasmLoadingTypes: [Object],
      Entry: [Object],
      EntryDescription: [Object],
      EntryDescriptionNormalized: [Object],
      EntryDynamic: [Object],
      EntryDynamicNormalized: [Object],
      EntryFilename: [Object],
      EntryItem: [Object],
      EntryNormalized: [Object],
      EntryObject: [Object],
      EntryRuntime: [Object],
      EntryStatic: [Object],
      EntryStaticNormalized: [Object],
      EntryUnnamed: [Object],
      Environment: [Object],
      Experiments: [Object],
      ExternalItem: [Object],
      ExternalItemFunctionData: [Object],
      ExternalItemValue: [Object],
      Externals: [Object],
      ExternalsPresets: [Object],
      ExternalsType: [Object],
      FileCacheOptions: [Object],
      Filename: [Object],
      FilenameTemplate: [Object],
      FilterItemTypes: [Object],
      FilterTypes: [Object],
      GeneratorOptionsByModuleType: [Object],
      GlobalObject: [Object],
      HashDigest: [Object],
      HashDigestLength: [Object],
      HashFunction: [Object],
      HashSalt: [Object],
      HotUpdateChunkFilename: [Object],
      HotUpdateGlobal: [Object],
      HotUpdateMainFilename: [Object],
      IgnoreWarnings: [Object],
      IgnoreWarningsNormalized: [Object],
      Iife: [Object],
      ImportFunctionName: [Object],
      ImportMetaName: [Object],
      InfrastructureLogging: [Object],
      JavascriptParserOptions: [Object],
      Layer: [Object],
      Library: [Object],
      LibraryCustomUmdCommentObject: [Object],
      LibraryCustomUmdObject: [Object],
      LibraryExport: [Object],
      LibraryName: [Object],
      LibraryOptions: [Object],
      LibraryType: [Object],
      Loader: [Object],
      MemoryCacheOptions: [Object],
      Mode: [Object],
      ModuleFilterItemTypes: [Object],
      ModuleFilterTypes: [Object],
      ModuleOptions: [Object],
      ModuleOptionsNormalized: [Object],
      Name: [Object],
      NoParse: [Object],
      Node: [Object],
      NodeOptions: [Object],
      Optimization: [Object],
      OptimizationRuntimeChunk: [Object],
      OptimizationRuntimeChunkNormalized: [Object],
      OptimizationSplitChunksCacheGroup: [Object],
      OptimizationSplitChunksGetCacheGroups: [Object],
      OptimizationSplitChunksOptions: [Object],
      OptimizationSplitChunksSizes: [Object],
      Output: [Object],
      OutputModule: [Object],
      OutputNormalized: [Object],
      Parallelism: [Object],
      ParserOptionsByModuleType: [Object],
      Path: [Object],
      Pathinfo: [Object],
      Performance: [Object],
      PerformanceOptions: [Object],
      Plugins: [Object],
      Profile: [Object],
      PublicPath: [Object],
      RawPublicPath: [Object],
      RecordsInputPath: [Object],
      RecordsOutputPath: [Object],
      RecordsPath: [Object],
      Resolve: [Object],
      ResolveAlias: [Object],
      ResolveLoader: [Object],
      ResolveOptions: [Object],
      ResolvePluginInstance: [Object],
      RuleSetCondition: [Object],
      RuleSetConditionAbsolute: [Object],
      RuleSetConditionOrConditions: [Object],
      RuleSetConditionOrConditionsAbsolute: [Object],
      RuleSetConditions: [Object],
      RuleSetConditionsAbsolute: [Object],
      RuleSetLoader: [Object],
      RuleSetLoaderOptions: [Object],
      RuleSetRule: [Object],
      RuleSetRules: [Object],
      RuleSetUse: [Object],
      RuleSetUseItem: [Object],
      ScriptType: [Object],
      SnapshotOptions: [Object],
      SourceMapFilename: [Object],
      SourcePrefix: [Object],
      StatsOptions: [Object],
      StatsValue: [Object],
      StrictModuleErrorHandling: [Object],
      StrictModuleExceptionHandling: [Object],
      Target: [Object],
      UmdNamedDefine: [Object],
      UniqueName: [Object],
      WarningFilterItemTypes: [Object],
      WarningFilterTypes: [Object],
      WasmLoading: [Object],
      WasmLoadingType: [Object],
      Watch: [Object],
      WatchOptions: [Object],
      WebassemblyModuleFilename: [Object],
      WebpackOptionsNormalized: [Object],
      WebpackPluginFunction: [Object],
      WebpackPluginInstance: [Object]
    },
    description: 'Options object as provided by the user.',
    type: 'object',
    additionalProperties: false,
    properties: {
      amd: [Object],
      bail: [Object],
      cache: [Object],
      context: [Object],
      dependencies: [Object],
      devServer: [Object],
      devtool: [Object],
      entry: [Object],
      experiments: [Object],
      externals: [Object],
      externalsPresets: [Object],
      externalsType: [Object],
      ignoreWarnings: [Object],
      infrastructureLogging: [Object],
      loader: [Object],
      mode: [Object],
      module: [Object],
      name: [Object],
      node: [Object],
      optimization: [Object],
      output: [Object],
      parallelism: [Object],
      performance: [Object],
      plugins: [Object],
      profile: [Object],
      recordsInputPath: [Object],
      recordsOutputPath: [Object],
      recordsPath: [Object],
      resolve: [Object],
      resolveLoader: [Object],
      snapshot: [Object],
      stats: [Object],
      target: [Object],
      watch: [Object],
      watchOptions: [Object]
    }
  },
  headerName: 'Webpack',
  baseDataPath: 'configuration',
  postFormatter: [Function: postFormatter]
}
```

</details>

This is my `next.config.js`:

```js
module.exports = require('@module-federation/next-transpile-modules')(['react-doorman'])({
  future: {
    webpack5: true
  }
})
```

When I do that, `yarn next build` does not work, and I get the error above. If I remove `react-doorman`, and instead use an empty array, it does work. 

I tried adding `{ resolveSymlinks: true }`, but no luck.

This also works:

```js
// next.config.js

module.exports = {
  future: {
    webpack5: true,
  },
}
```

When I run `yarn list webpack`, I see this:

```sh
yarn list webpack
yarn list v1.22.5
warning Resolution field "@theme-ui/css@0.4.0-rc.1" is incompatible with requested version "@theme-ui/css@^0.4.0-rc.13"
warning Filtering by arguments is deprecated. Please use the pattern option instead.
└─ webpack@4.43.0
✨  Done in 1.11s.
```

I'm not sure why it's not v5, since Next.js says it ships with v5 🤷🏼‍♂️

My real app has a much larger `next.config.js` file. I stripped it down in a fresh project to try to debug this issue. I am actually transpiling 20+ modules in my real app, so I need to use `next-transpile-modules`, in addition to other plugins.

Do you have any idea of what I might need to do to fix this? Or do you need any extra info from me to get a better idea? Really appreciate your help. Thanks so much!

**Comment by nandorojo** [comment]
Ah, sorry, I forgot to upgrade `@module-federation/next-transpile-modules` to `6.4.0`. Upgrading solved this. The final problem I'm having is that the new version isn't transpiling my locally-scoped packages from my monorepo. This problem seems a bit different, so maybe I should open a separate issue.

```sh
Error: next-transpile-modules - an unexpected error happened when trying to resolve "@beatgig/analytics"
Error: Can't resolve '@beatgig/analytics' in '/Users/fernandorojo/Developer/madison-green/beatgig-playground/applications/speed'
    at getPackageRootDirectory
```

This happens with the new version of `@module-federation/next-transpile-modules`, as well as `next-transpile-modules`, regardless of enabling webpack5 or not. So maybe this issue belongs on `next-transpile-modules`.

Is there any reason to continue using your fork, or has the main repo solved the open problems you fixed @ScriptedAlchemy?
